arangod/Agency/Job.cpp

```diff
@@ -288,7 +317,7 @@ std::vector<Job::shard_t> Job::clones(Node const& snapshot, std::string const& d
     auto const otherCollection = colptr.first;
 
     if (otherCollection != collection && col.has("distributeShardsLike") &&  // use .has() form to prevent logging of missing
-        col.hasAsSlice("distributeShardsLike").first.copyString() == collection) {
+        col.hasAsBuilder("distributeShardsLike").first.slice().copyString() == collection) {
```

Comment:
This will throw if "distributeShardsLike" is there but is not a string. Maybe add a test.

Reply:
Agree